techhive-ui/src/components/ChatContainer.jsx

```diff
@@ -0,0 +1,380 @@
+import React, { useState, useEffect, useRef, useContext } from "react";
+import styled from "styled-components";
+import ChatInput from "./ChatInput.jsx";
+import { v4 as uuidv4 } from "uuid";
+import axios from "axios";
+import { sendMessageRoute, receiveMessageRoute} from "../utils/APIRoutes";
+import { UserContext } from '../UserContext.jsx';
+
+export default function ChatContainer({ currentChat, socket }) {
+    const { user } = useContext(UserContext);
+    const [messages, setMessages] = useState([]);
+    const scrollRef = useRef();
+    const [receivedMessages, setReceivedMessages] = useState([]);
+    const [searchQuery, setSearchQuery] = useState("");
+    const [filteredMessages, setFilteredMessages] = useState([]);
+    
+
+useEffect(() => {
+    const fetchMessages = async () => {
+        try {
+          const response = await axios.post(receiveMessageRoute, {
+            from: user.username,
+            to: currentChat.username,
+          });
+          const messagesFromServer = response.data;
+      
+          // Filter messages where the sender is the current user (fromSelf)
+          const sentMessages = messagesFromServer
+            .filter((msg) => msg.sender === user.username)
+            .map((msg) => ({
+              ...msg,
+              timestamp: parseISOString(msg.createdAt),
+              originalTimestamp: msg.createdAt,
+            }));
+      
+          // Filter messages where the sender is the current chat user (received messages)
+          const receivedMessages = messagesFromServer
+            .filter((msg) => msg.sender === currentChat.username)
```

Comment:
what's the difference between currentChat.username and user.username above? 

maybe the naming can be improved here